crates/sui-config/src/utils.rs

```diff
@@ -49,3 +49,26 @@ pub fn available_local_socket_address() -> std::net::SocketAddr {
         .parse()
         .unwrap()
 }
+
+pub fn udp_multiaddr_to_socket_address(
+    multiaddr: &multiaddr::Multiaddr,
+) -> Option<std::net::SocketAddr> {
+    use multiaddr::Protocol;
+    let mut iter = multiaddr.iter();
+
+    match (iter.next(), iter.next()) {
```

Comment:
is this safe or could it ever crash if we have iter without that many elements?

Reply:
Yep this is safe, if an iterator is empty it should return `None`